### ISSUE TITLE
SEA - user data with not signed

### ISSUE THREAD

**Issue by rogowski** [amark/gun#616]
Hi guys!

Somenthig wrong is hapennig with my user data (SEA signed), I guess.

```
~VBYSNad5BPMwtF3-WdAc8AcTKna1_z-Cms8TAvklMB4.ybkGu54ov4e7th0-jxGh6UsQ92RWeuTnHnr9o1WbdGY
    alias: "SEA{"m":"\"myusername\"","s":"ÂýJ\r\u0016d\u001c\rfünZ\t^U±²\u001e!.....
    auth: "SEA{"m":"{\"ek\":\"SEA{\\\"ct\\\":\\\"Q\\\\u0010ÀÑ\\\\u0015pÛÁÏ.ùÒ!øI\.....
    category: "SEA{"m":"{\"#\":\"jml0nvw6cCNIJKJ1TcrJ~VBYSNad5BPMwtF3-WdAc8AcTKna.....
    data: {#: "jmla7120iNNuZwWlhQH}    <<<<<---------- NOT SIGNED. ITS AN OBJECT AND NOT A SEA STRING LIKE OTHERS'SEA{....'

    epub: "SEA{"m":"\"gp7Lh5ewZ0c9MzZQldLSeB0m1bm2xETFAV0NX-xckU8.-ycqgAILbs_MrDw.....
    pub: "VBYSNad5BPMwtF3-WdAc8AcTKna1_z-Cms8TAvklMB4.ybkGu54ov4e7th0-jxGh6UsQ92R.....

```

When I try to signin, this nos signed data crash.

Has this ever happened to you?

## If It's a bug, there's two problems to solve in my opinion.
 1) **MUST not crash my auth** and the data must be accessible. (ignore data, delete, prevent errors...) 
 2) Discover why it's not signed(Occurs eventually).

###### My code

```javascript
gun.on('auth', function() {

  let public_node = gun.get('sinalization').get('001').get('request').on((v, pk) => {
    if (v === null) {return;}

    let d = user.get('data').get(pk).put({}, (ack) => {
      gun.user(pk).get('profile').once(function(profile, k) {
            if (profile && profile._) {
              delete profile._;
            }
            let data = {};
            if (profile) {
              data = Gun.obj.copy(profile);
            } else {
              console.log('[WARN] profile', pk);
            }
            d.put(data);
      });


------ IN OTHER BROWSER, WITH OTHER USER ------

gun.get('sinalization').get('001').get('request').put('ABC');

```

#### SEA refuses the data unsigned

 here: https://github.com/amark/gun/blob/f1ff3b005cf3348aefb51e9b243a45ec2c9c456c/sea.js#L1563

##### ... in the browser console...
```
sea.js:451
    TypeError: Cannot read property 'byteLength' of undefined
    at Function.from (sea.js:127)
    at Function.SEA.verify.SEA.verify (sea.js:443)

sea.js:1648 
    NO! Unverified data. {~VBYSNad5BPMwtF3-WdAc8AcTKna1_z-.......
```

To continue with my data, I added the above line in SEA. So the authentication not fail and I can `.put(null)` to not signed data.
```
L1563 if (typeof val !== 'string') { check['user'+soul+key] = 0; each.end({ok: 0}); return; }
L1563 SEA.verify(val, pub, function(data){ var rel, tmp;
```

**Comment by rogowski** [comment]
How to replicate:
```
user.create('usr', 'pass');
user.auth('usr', 'pass');
user.get('aaa').get('bbb').get('ccc').put({});
```
The problem is with empty objects.

**Comment by amark** [comment]
@rogowski hmm, @mmalmi found something similar with `.set(` because the code internally goes to gun root and saves (which puts it outside the user context) so I need to fiddle with that.

**Comment by rogowski** [comment]
Removing the line https://github.com/amark/gun/blob/master/sea.js#L1143 works like expected.

And for nested objects not signed like `user.get({name:"Bob", things: {a:1,b:2}})` (things's soul is not created with the pubkey and not signed), changing the line https://github.com/amark/gun/blob/master/gun.js#L1766 to
`item = gun.back(-1).get(soul = soul || Gun.node.soul(item) || gun.back('opt.uuid')()).put(item, null, {via: gun});` it works.

These two changes seem to work fine.

**Comment by rogowski** [comment]
Fixed version : "0.2020.421"
